packages/create-bison-app/template/src/env.mjs

```diff
@@ -0,0 +1,96 @@
+import { z } from 'zod';
+
+const nodeEnv = process.env.NODE_ENV || 'development';
+
+/**
+ * Specify your server-side environment variables schema here. This way you can ensure the app isn't
+ * built with invalid env vars.
+ */
+const server = z.object({
+  APP_ENV: z.enum(['development', 'test', 'production']),
+  APP_URL: z.string(),
+  DATABASE_URL: z.string().url(),
+  DEBUG: z.boolean().optional(),
+  FC_GIT_COMMIT_SHA: z.string().optional(),
+  IS_CI: z.boolean().optional(),
+  NEXTAUTH_SECRET:
+    process.env.NODE_ENV === 'production' ? z.string().min(1) : z.string().min(1).optional(),
+  NEXTAUTH_URL: z.string().min(1),
+  NODE_ENV: z.enum(['development', 'test', 'production']),
+  PORT: z.number().int().positive(),
+  RENDER_EXTERNAL_URL: z.string().optional(),
+  SHOULD_MIGRATE: z.boolean(),
+});
```

Comment:
I’ve run into a lot of annoying issues using zod to validate config in this way and on a couple projects have ended up just scrapping it. A couple reasons why:

- You have to coerce to get booleans and numbers from env strings. This doesn't work the way you want by default (e.g. `"false"` will be coerced as `true`). You can write transformation functions for zod which works ok, but it isn't idea.

- Sometimes config for different stages have different requirements.

- There was another reason and I can't remember it.

 I’d avoid running it against config in this way. There are a few alternatives:
 
 1. The way we're doing config on Tachus (e.g. those `envToString`, etc. functions). This has worked great for me on a number of projects.
 
 2. Run a zod schema check on `process.env` itself to just verify that required values are provided and nothing more.
 
 3. Create your `config` variable using option no. 1 above, but then run a zod schema check against that.

Reply:
I think this is a good pattern for config:

https://gist.github.com/cullylarson/d710be62559c593c0f7dd6680e22daca

Reply:
These are all great comments Cully. 
Reading through - I'm fine not moving along with Zod at the moment.